### PR TITLE
Refactor default dir index

### PR DESCRIPTION
**Pull request by pahud**
1. rename to `DefaultDirIndex` construct class name
2. apply solution ID and template description
3. refactor the directory
4. move the demo to demo directory
5. add unit testing and integ testing
6. move the test directory to` src/__tests__`

### CODE REVIEW

File: src/demo/default-dir-index/index.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+fs.mkdirSync(path.join(__dirname, 'a/b/c'), {
+  recursive: true,
+});
+fs.writeFileSync(path.join(__dirname, 'a/b/c/index.html'), '<h1>Hello CDK!</h1>');
+// Put demo Object to Bucket.
+new BucketDeployment(rewriteUriDemo, 'Deployment', {
+  sources: [Source.asset(path.join(__dirname, './'))],
+  destinationBucket: bucket,
+  retainOnDelete: false,
+});
```
Comment (pahud) [inline src/demo/default-dir-index/index.ts:34]: let's auto generate the demo directories for deployment like this @guan840912

File: src/extensions.ts [hunk 2]
```diff
@@ -167,24 +172,54 @@ export interface CustomProps {
...
+   *
+   * @default ''
+   */
+  readonly templateDescription?: string;
 }
 /**
  * Custom extension sample
  */
-export class Custom implements IExtensions {
+export class Custom extends cdk.NestedStack implements IExtensions {
```
Comment (pahud) [inline src/extensions.ts:191]: we need a nested stack to hold the solution id and template description.

File: src/__tests__/integ.default-dir-index.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                'Outputs.demostackDefaultDirIndexDemoFuncVerDefaultDirIndexDemo2C78B9F8Ref',
+              ],
+            },
+          },
+        ],
+        ViewerProtocolPolicy: 'redirect-to-https',
+      },
+      DefaultRootObject: 'index.html',
+    },
+  });
```
Comment (pahud) [inline src/__tests__/integ.default-dir-index.test.ts:80]: added tests here

File: src/demo/default-dir-index/index.ts (hunk 1 above)
Comment (guan840912) [reply src/demo/default-dir-index/index.ts:34]: woo! ,this is awesome

File: src/__tests__/integ.default-dir-index.test.ts (hunk 3 above)
Comment (guan840912) [reply src/__tests__/integ.default-dir-index.test.ts:80]: LGTM!!

**Review by guan840912 — APPROVED** [review]
LGTM

**Pull request merged**